src/main.rs

```diff
@@ -150,18 +155,19 @@ impl ClippyCmd {
 
     fn into_std_cmd(self) -> Command {
         let mut cmd = Command::new("cargo");
-        let clippy_args: String = self
-            .clippy_args
-            .iter()
-            .map(|arg| format!("{}__CLIPPY_HACKERY__", arg))
-            .collect();
 
         cmd.env(self.path_env(), Self::path())
             .envs(ClippyCmd::target_dir())
-            .env("CLIPPY_ARGS", clippy_args)
             .arg(self.cargo_subcommand)
             .args(&self.args);
 
+        // HACK: pass Clippy args to the driver *also* through RUSTFLAGS.
+        // This guarantees that new builds will be triggered when Clippy flags change.
+        if let (Some(clippy_args), Some(rustflags)) = (self.clippy_args, self.rustflags) {
+            cmd.env("CLIPPY_ARGS", clippy_args);
+            cmd.env("RUSTFLAGS", rustflags);
+        }
```

Comment:
~~Shouldn't we also pass Clippy args when `self.rustflags` is ``None``?~~

Ah I think I get the intention of this now. If ``self.clippy_args`` is ``Some``, then we copy them to ``self.rustflags`` in the constructor. So `self.clippy_args.is_some() => self.rustflags.is_some()`.

But, we only use ``ClippyCmd::rustflags`` here. Couldn't we just pass the ``rustflags`` into this function and extend them here? That would simplify this a lot, IMO:

```suggestion
        if let Some(clippy_args) = self.clippy_args {
            cmd.env("CLIPPY_ARGS", clippy_args.clone());
            cmd.env("RUSTFLAGS", rustflags.map_or(clippy_args, |flags| format!({} {}, clippy_args, flags)));
        }
```

Reply:
Yep, I added two different variables in the struct to make it easy to test. If I add it at this point, we won't have the `ClippyCmd` anymore.

There's a nightly feature to [retrieve the environment vars](https://doc.rust-lang.org/std/process/struct.Command.html#method.get_envs) set in a `std::process::Command`, I can use that instead in the tests.

Reply:
> There's a nightly feature to [retrieve the environment vars](https://doc.rust-lang.org/std/process/struct.Command.html#method.get_envs) set in a `std::process::Command`, I can use that instead in the tests.

Let's do that then. I think it improves the readability of the code quite a bit. Less spaghetti'ish.